src/storage/kv.js

```diff
@@ -0,0 +1,184 @@
+const api = require('../api');
+const { getNamespaceByName } = require('./storage');
+
+/**
+ * Gets namespaces associated with the account
+ * @param {*} options
+ */
+async function getKeys({ accountId, nameSpace } = {}) {
+  const account = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!account) {
+    throw 'You must provide an account ID';
+  }
+
+  const nameSpaceId = await getNamespaceByName({
+    accountId: account,
+    name: nameSpace
+  });
+
+  if (!nameSpaceId) {
+    throw 'Name space does not exist';
+  }
+
+  let result = await api.cfApiCall({
+    url: `/accounts/${account}/storage/kv/namespaces/${nameSpaceId}/keys`,
+    method: 'GET'
+  });
+
+  if (result.success && result.result) {
+    result = result.result;
+  }
+
+  return result;
+}
+
+/**
+ * Gets namespaces associated with the account
+ * @param {*} options
+ */
+async function getAllKeysAndValues({ accountId, nameSpace } = {}) {
+  const account = accountId || process.env.CLOUDFLARE_ACCOUNT_ID;
+
+  if (!account) {
+    throw 'You must provide an account ID';
+  }
+
+  const keys = await getKeys({
+    accountId: account,
+    nameSpace: nameSpace
+  });
+
+  const result = [];
+  for (const key of keys) {
+    const value = await getKeyValue({
```

Comment:
This is really inefficient since it looks up the namespace in getKeyValue every time. If you want to provide the "look up namespace automatically" functionality you should look it up once and pass it into getKeyValue

Reply:
GOod catch